### PR TITLE
Fix gate failing randomly

### PR DESCRIPTION
**Pull request by scarf005**
#### Summary

SUMMARY: Bugfixes "Fix mechanical gates randomly failing to close"

#### Purpose of change

- `gates::open_gate` fails because `forced_door_closing` returns `false`
- `forced_door_closing` randomly fails because it checks room for pushback regardless of whether there's item/monster
https://github.com/cataclysmbnteam/Cataclysm-BN/blob/6b7f879b6e6247d628d4ba2ae4680d8d86d770d8/src/game.cpp#L5117-L5121

#### Describe the solution

- 1724e0b7ccf051646cbc286e9bf2091dbefa0f17, 1724e0b7ccf051646cbc286e9bf2091dbefa0f17: lazy evaluate pushback check until `pushback()` and item moving is actually done
- 88c3ab07fb54c2db4dcda08b3a2c65285f5b1dfa: renamed `open_gate` and releated function name to `toggle_gate`
- 70708db39bb4d2a6150fba1641c49eeb5529ce5d, 29e5683b3d4120edab58786eeb2e0cafae1bdef8: removed duplicate points

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -5196,9 +5202,9 @@ bool game::forced_door_closing( const tripoint &p, const ter_id &door_type, int
         }
     }
 
-    m.ter_set( point( x, y ), door_type );
-    if( m.has_flag( "NOITEM", point( x, y ) ) ) {
-        map_stack items = m.i_at( point( x, y ) );
+    m.ter_set( p, door_type );
+    if( m.has_flag( "NOITEM", p ) ) {
+        map_stack items = m.i_at( p );
         for( map_stack::iterator it = items.begin(); it != items.end(); ) {
```
[the remark this answers is not in the document]
Comment (Coolthulhu) [reply src/game.cpp:5208]: Nope. Later you have `it =`.
This style is used when the iterator is altered in the loop, usually with `erase`.

File: src/game.cpp [hunk 2]
```diff
@@ -5100,27 +5100,27 @@ void game::exam_vehicle( vehicle &veh, const point &c )
...
-        // can't pushback any creatures anywhere, that means the door can't close.
-        return false;
-    }
-    const bool can_see = u.sees( tripoint( x, y, p.z ) );
-    player *npc_or_player = critter_at<player>( tripoint( x, y, p.z ), false );
+    const auto get_random_point = [&]() -> tripoint {
+        if( auto pos = random_point( m.points_in_radius( p, 2 ), valid_location ) )
+        {
+            return  p * 2 - ( *pos );
+        } else
```
Comment (Coolthulhu) [inline src/game.cpp:5110]: Shouldn't have a newline here or after `if(...)`.

File: src/game.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (scarf005) [reply src/game.cpp:5208]: alright then

File: src/game.cpp (hunk 2 above)
Comment (scarf005) [reply src/game.cpp:5110]: it's astyle problem. astyle does not understands lambda and makes inserts newlines to in expression

**Pull request merged**